src/Entity/PropertyId.php

```diff
@@ -48,7 +48,7 @@ private function assertValidIdFormat( $idSerialization ) {
 	 * @return int
 	 */
 	public function getNumericId() {
-		return (int)substr( $this->serialization, 1 );
+		return (int)substr( $this->getSerialization(), 1 );
```

Comment:
Actually, getNumericId should fail for foreign IDs. Using the numeric ID of a foreign ID locally can never be right.